# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/tests/window_composition_text_querycontent.xhtml
# Commit: 5f90ad3d8781
# Full Hash: 5f90ad3d87818a0810f74bde3bfc71d485eea801
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-02-08 17:32:27
# Regressor Bug: 1746104
# File Overlap Count: 1
# Description:
#   Bug 1746104 - part 5-1: Get rid of `WidgetQueryContentEvent::Reply::mHasSelection` r=m_kato
#   
#   It's intended to indicate whether the selection is collapsed or not, but it can
#   be referred by other members, there is no reasonable user and the name makes
#   developers confused.
# ==============================================================================

diff -r dc81d96d0adb -r 5f90ad3d8781 widget/tests/window_composition_text_querycontent.xhtml
--- a/widget/tests/window_composition_text_querycontent.xhtml	Mon Feb 07 22:33:37 2022 +0000
+++ b/widget/tests/window_composition_text_querycontent.xhtml	Mon Feb 07 22:33:37 2022 +0000
@@ -203,10 +203,33 @@
                                ": synthesizeQuerySelectedText " + aID)) {
     return false;
   }
-  is(selectedText.offset, aExpectedOffset,
-     aMessage + ": selection offset is wrong " + aID);
-  is(selectedText.text, aExpectedText,
-     aMessage + ": selected text is wrong " + aID);
+  if (aExpectedOffset === null) {
+    todo_is(
+      selectedText.notFound,
+      true,
+      `${aMessage}: selection should not be found ${aID}`
+    );
+    return selectedText.notFound;
+  }
+
+  is(
+    selectedText.notFound,
+    false,
+    `${aMessage}: selection should be found ${aID}`
+  );
+  if (selectedText.notFound) {
+    return false;
+  }
+  is(
+    selectedText.offset,
+    aExpectedOffset,
+    `${aMessage}: selection offset should be ${aExpectedOffset} ${aID}`
+  );
+  is(
+    selectedText.text,
+    aExpectedText,
+    `${aMessage}: selected text should be "${aExpectedText}" ${aID}`
+  );
   return selectedText.offset == aExpectedOffset &&
          selectedText.text == aExpectedText;
 }
@@ -4551,18 +4574,54 @@
 
   // #1
   contenteditable.innerHTML = "<br/>a";
-  selection.setBaseAndExtent(contenteditable.firstChild, 0, contenteditable.lastChild, 1);
-  checkSelection(0, kLF + "a", "runQuerySelectionEventTest #1, \"" + contenteditable.innerHTML + "\"");
+  selection.setBaseAndExtent(
+    contenteditable.firstChild,
+    0,
+    contenteditable.lastChild,
+    1
+  );
+  checkSelection(
+    0,
+    `${kLF}a`,
+    `runQuerySelectionEventTest #1, "${contenteditable.innerHTML}"`
+  );
 
   // #2
   contenteditable.innerHTML = "<p></p><p>abc</p>";
-  selection.setBaseAndExtent(contenteditable.firstChild, 0, contenteditable.lastChild.firstChild, 1);
-  checkSelection(kLFLen, kLF + "a", "runQuerySelectionEventTest #2, \"" + contenteditable.innerHTML + "\"");
+  selection.setBaseAndExtent(
+    contenteditable.firstChild,
+    0,
+    contenteditable.lastChild.firstChild,
+    1
+  );
+  checkSelection(
+    kLFLen,
+    `${kLF}a`,
+    `runQuerySelectionEventTest #2, "${contenteditable.innerHTML}"`
+  );
 
   // #3
   contenteditable.innerHTML = "<p>abc</p><p>def</p>";
-  selection.setBaseAndExtent(contenteditable.firstChild, 0, contenteditable.lastChild.firstChild, 1);
-  checkSelection(kLFLen, "abc" + kLF + "d", "runQuerySelectionEventTest #3, \"" + contenteditable.innerHTML + "\"");
+  selection.setBaseAndExtent(
+    contenteditable.firstChild,
+    0,
+    contenteditable.lastChild.firstChild,
+    1
+  );
+  checkSelection(
+    kLFLen,
+    `abc${kLF}d`,
+    `runQuerySelectionEventTest #3, "${contenteditable.innerHTML}"`
+  );
+
+  // #4
+  contenteditable.innerHTML = "<p>abc</p>";
+  selection.removeAllRanges();
+  checkSelection(
+    null,
+    null,
+    `runQuerySelectionEventTest #4, "${contenteditable.innerHTML}"`
+  );
 }
 
 function runQueryIMESelectionTest()
